src/plugins/intel_gpu/src/plugin/ops/matmul.cpp

```diff
@@ -190,71 +190,86 @@ static void CreateMatMulOp(Program& p, const std::shared_ptr<ngraph::op::v0::Mat
-    //                                           target_format,
-    //                                           targetDatatype,
-    //                                           std::vector<float>(),
-    //                                           cldnn::reorder_mean_mode::subtract,
-    //                                           op->get_friendly_name());
-
-    //         p.AddPrimitive(reorderPrim);
-    //         p.AddInnerPrimitiveToProfiler(reorderName, layerName, op);
-
-    //         inputPrimitives[i] = reorderName;
-    //     }
-
-    //     // Reshape input if they differ or gemm specific shape matches default one
-    //     if (input_rank != output_rank || input_rank < 4) {
-    //         auto reshapeName = layerName + "_cldnn_in" + std::to_string(i) + "_reshape";
-
-    //         // Extend input dimensions by prepending ones
-    //         if (input_rank == 1) {
-    //             // One-dimensional tensors unsqueezing is applied for each input independently.
-    //             // The axes inserted in this step are not included in the output shape.
-    //             // * If rank of the **first** input is equal to 1, it is always unsqueezed to 2D tensor **row vector** (regardless of `transpose_a`)
-    //             // by adding axes with size 1 at ROW_INDEX_DIM, to the **left** of the shape. For example `[S]` will be reshaped to `[1, S]`.
-    //             // * If rank of the **second** input is equal to 1, it is always unsqueezed to 2D tensor **column vector** (regardless of `transpose_b`)
-    //             // by adding axes with size 1 at COL_INDEX_DIM, to the **right** of the shape. For example `[S]` will be reshaped to `[S, 1]`.
-    //             bool transpose = false;
-    //             if (i == 0) {
-    //                 transpose = op->get_transpose_a();
-    //                 input_pshape.insert(input_pshape.begin(), 1);
-    //             } else {
-    //                 transpose = op->get_transpose_b();
-    //                 input_pshape.insert(input_pshape.end(), 1);
-    //             }
-    //             // Specs says that shapes must be unsqueezed regardless of tranpose flag, but primitive implementation always respects transposes
-    //             // so we have to swap dimensions correspondingly to have consistent shapes.
-    //             if (transpose) {
-    //                 std::swap(input_pshape[0], input_pshape[1]);
-    //             }
-    //         }
-    //         if (input_rank < output_rank)
-    //             input_pshape.insert(input_pshape.begin(), output_rank - input_rank, 1ul);
-
-    //         auto reshapePrim = cldnn::reshape(reshapeName, inputPrimitives[i], input_pshape, op->get_friendly_name());
-
-    //         p.AddPrimitive(reshapePrim);
-    //         p.AddInnerPrimitiveToProfiler(reshapeName, layerName, op);
-
-    //         inputPrimitives[i] = reshapeName;
-    //     }
-    // }
+    auto output_pshape = op->get_output_partial_shape(0);
```

Comment:
I think the intention of the original Vladimir's removal of this part is that because gemm is now using ngraph's shape inference, so no need to add reshape or reorder to 4d... Could you let me know why the original vladimir's impl failed for the target gemms ?

Reply:
As we discussed this seperately, I reverted this commit and moved pre/post-processing for in/output of gemm to premitive_impl level (https://github.com/andrew-k-park/openvino/commit/c769d7db19b661848a6e065c30e6427ac07b80ea)

Reply:
left comment to the commit 
https://github.com/andrew-k-park/openvino/commit/c769d7db19b661848a6e065c30e6427ac07b80ea

Reply:
As discussed, this part will be handled through a separate task.